src/clj/rems/workflow/dynamic.clj

```diff
@@ -27,25 +27,120 @@
     :application.state/submitted
     #_:application.state/withdrawn}) ; TODO withdraw support?
 
-(def CommandTypes
-  #{#_:application.command/accept-license
-    #_:application.command/require-license
-    :application.command/accept-invitation
-    :application.command/add-member
-    :application.command/invite-member
-    :application.command/approve
-    :application.command/close
-    :application.command/comment
-    :application.command/decide
-    :application.command/reject
-    :application.command/request-comment
-    :application.command/request-decision
-    :application.command/remove-member
-    :application.command/return
-    :application.command/save-draft
-    :application.command/submit
-    :application.command/uninvite-member
-    #_:application.command/withdraw})
+(s/defschema CommandInternal
+  {:actor UserId
+   ;; TODO: tests use non-DateTime times
+   :time s/Any})
+
+(s/defschema CommandBase
+  {:application-id s/Int})
+
+(s/defschema SaveDraftCommand
+  (assoc CommandBase
+         :type (s/eq :application.command/save-draft)
+         :field-values s/Any
+         :accepted-licenses s/Any))
+
+(s/defschema SubmitCommand
+  (assoc CommandBase
+         :type (s/eq :application.command/submit)))
+
+(s/defschema ApproveCommand
+  (assoc CommandBase
+         :type (s/eq :application.command/approve)
+         :comment s/Str))
+
+(s/defschema RejectCommand
+  (assoc CommandBase
+         :type (s/eq :application.command/reject)
+         :comment s/Str))
+
+(s/defschema ReturnCommand
+  (assoc CommandBase
+         :type (s/eq :application.command/return)
+         :comment s/Str))
+
+(s/defschema CloseCommand
+  (assoc CommandBase
+         :type (s/eq :application.command/close)
+         :comment s/Str))
+
+(s/defschema RequestDecisionCommand
+  (assoc CommandBase
+         :type (s/eq :application.command/request-decision)
+         :deciders [UserId]
+         :comment s/Str))
+
+(s/defschema DecideCommand
+  (assoc CommandBase
+         :type (s/eq :application.command/decide)
+         :decision s/Keyword ;; TODO: (s/enum :approved :rejected), validated now in the command handler
```

Comment:
Could it be done here instead and removed from the command handler? It should require only changing the test's assertion because the error message will be different when it comes from schema validation.